### ISSUE TITLE
Add OLTP test case with replication turned on

### ISSUE THREAD

**Issue by jkosh44** [cmu-db/noisepage#1554]
# Feature Request
## Summary
It would be nice if, during the performance stage of CI, we ran an OLTP benchmark with replication turned on. This way we can easily track improvements (or degradations) to replication related performance.

## Solution
We'll need to enhance the OLTP python scripts to be able to optionally start a replica node before running an OLTP Benchmark. Then we'll add some configurations and a Jenkins step to run during CI.

**Comment by lmwnshn** [comment]
> We'll need to enhance the OLTP python scripts to be able to optionally start a replica node before running an OLTP Benchmark. Then we'll add some configurations and a Jenkins step to run during CI.

I don't think you want to push the replica-testing logic into the OLTPBenchmark runner, so it probably makes more sense to

1. OLTPBenchmark runner needs to be able to assume the DBMS is already started (right now, it always wants to own it)
2. OLTPBenchmark runner should just connect to the DBMS and create/load/execute, separation of concerns
3. For your sanity, you may want to leave the glue to CI instead of trying to do it in Python, i.e.,

```
// 1. start noisepage and replicas from Jenkins
// 2. run oltpbench
// 3. run replica consistency tests
```

Though if you do expose a nice API in Python, that would be very helpful for eg `self_driving` stuff

**Comment by jkosh44** [comment]
@lmwnshn If I'm understanding correctly this would involve two separate tasks

1. Decouple the OLTPBench tests from the database instance (this isn't necessarily related to replication).
2. Add OLTPBench test cases that use replication in CI.

I'll probably try and tackle these separately once the semester is over.
